cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/lineage/field/FieldLineageInfo.java

```diff
@@ -126,6 +134,7 @@ public FieldLineageInfo(Collection<? extends Operation> operations) {
   public FieldLineageInfo(Collection<? extends Operation> operations, boolean computeSummaries) {
     LOG.trace("Received field lineage operations {}", GSON.toJson(operations));
     this.operations = new HashSet<>(operations);
+    this.droppedFields = new HashMap<>();
     computeAndValidateFieldLineageInfo(operations);
     this.checksum = computeChecksum();
     if (computeSummaries) {
```

Comment:
Here after we computeSummaries, we should remove the EPF key in the incoming summary if there is drop operation, as EPF is not an actual endpoint and it should have any incoming fields.